content/articles/diogogpinto-filament-themes-full-guide.md

```diff
@@ -0,0 +1,333 @@
+        './vendor/awcodes/filament-curator/resources/**/*.blade.php',
+    ],
+}
+```
+
+Additionally, in the case of the `Curator` plugin, the author requires you to import some CSS files in your `theme.css`. This can be done, by editing the file `resources/css/filament/admin/theme.css` and making it look like below:
+
+```css
+@import '/vendor/filament/filament/resources/css/theme.css';
+@import '/node_modules/cropperjs/dist/cropper.css';
+@import '/vendor/awcodes/filament-curator/resources/css/plugin.css';
+
+@config 'tailwind.config.js';
+```
+
+This last step imports all custom classes needed by the plugin to render correctly. It imports the class, puts into our panel's unique CSS file and minimizes it all for greater performance.
+
+#### Running the build process
+
+When installing plugins and editing any theme related files, always remember to run `npm run build` when you're finished to compile all the assets.
+
+## Customizing the theme
+
+Laravel, and Filament being Laravel based, offer multiple solutions to the same problem. So, we will address two ways of building your own Filament look and feel, the recommended way and the not-so-recommended way. Let's start with the latter.
+
+### Before starting the customization process
+
+You should start the `npm run dev` process in the root of your project. This process watches for changes in the files registered in the `vite.config.js` (and `tailwind.config.js` by extension) and automatically builds a temporary CSS file to render the changes on file save.
+
+> [!note]
+> You still need to run `npm run build` when you're done customizing your theme
+
+### The recommended way to customize your theme
+
+The Filament panels core offers you a list of classes. Every component of the panel layout, including forms, infolists and tables, were wrapped in custom classes to make it possible, for any developer, to customize the panel look and feel. Let's look at the following example:
+
+![Breadcrumb Example](/images/content/articles/diogogpinto-filament-themes-full-guide/breadcrumb-example.webp)
+
+Looking at the breadcrumb component with Chrome's builtin Developer Tools > Inspect Element, you can see that the `<nav>`na tag contains the following classes:
+
+```html
+<nav class="fi-breadcrumbs mb-2 hidden sm:block">
+```
+
+The `.fi-breadcrumbs` class is inserted to allow developers to further optimize the breadcrumbs' look and feel. By default, the class has no CSS properties defined in itself, it's only there for the developer to apply custom styles.
+
+#### Customizing the CSS
+
+Picking up in the above example of `.fi-breadcrumbs`, we can now apply custom CSS classes on our `theme.css`. You can either use Tailwind utility classes or just write vanilla CSS. Let's say you want to override the default margin `mb-2`. Inputting the below CSS will yield the exact same results:
+
+```css
+.fi-breadcrumbs {
+    @apply mb-4;
+}
+
+.fi-breadcrumbs {
+    margin-bottom: 1rem;
+}
```

Comment:
I actually think this isn't _necessarily_ always going to work.

I believe that, specifically referencing the bottom margin, we won't reliably be able to override the CSS styling. This is because there is already a `mb-2` class in the class string, so when our styles get applied via `.fi-breadcrumbs`, `mb-2` and our `mb-4` will (theoretically) have the same specificity. If _that_ is the case, whichever class is defined further down in the cascade with take precedence.

I might be wrong here, but regardless it may be worth changing this example from an "overriding" example to an "adding" example. Something like adding a top padding, border, etc.

@zepfietje (or anyone from Core), let me know if I'm completely incorrect here.

Reply:
@alexandersix just wondering, given that the custom classes are always defined after TW classes, wouldn't they always take precedence? Or is there a case where that may not be true?

I may be missing something

Reply:
If you’re overriding an existing style, the user will need to use `!important` or the `!` prefix in the Tailwind classes.

For example, if we’re changing the sidebar color, this code won’t work:

```css
.fi-sidebar {
  @apply bg-white;
}
```
To ensure it takes precedence over Filament’s predefined styles, we need to mark the class as important:

```css
.fi-sidebar {
  @apply !bg-white;
}
```